### DISCUSSION TITLE
Rendering quarto project with multiple conda environments

### DISCUSSION THREAD

**Discussion by royfrancis** [quarto-dev/quarto-cli#10606] [Q&A] [answered]
### Description

I have a quarto project (website). There are a few different pages with R and Python code chunks. Some pages use conda env A while other pages use conda env B. When I have a single conda env, I would just do:

```
conda activate bla
quarto render
```

And that would generate the whole website `_site`. I am wondering how to go about rendering the whole website when different pages use different conda environments? One possibility is to render page by page.

```
conda activate bla1
quarto render x.qmd
quarto render y.qmd

conda activate bla2
quarto render p.qmd
quarto render q.qmd
```

Is there a better way to do this? Another detail is that some files are ipynb with python code while others are qmd files with R code.

**Comment by mcanouil** [comment, marked as answer]
Unfortunately, no. Incremental rendering and freeze for computation is the current way to go.

**Reply by royfrancis** [reply]
Ok. thanks! My current strategy is to render the whole website with one conda env (while manually disabling code execution in documents that use alternate env). This sets up the home, menus, sidebar and everything correctly. And then enable code execution and re-render some documents separately in alternate env.

I was wondering if there would be some solution using pre-render scripts to run 'conda activate bla' before each document or something.

**Reply by mcanouil** [reply]
> I was wondering if there would be some solution using pre-render scripts to run 'conda activate bla' before each document or something.

pre-render is a project wide option, not a per-file configuration.
